components/LogWorkout/LogWorkout.jsx

```diff
@@ -76,6 +86,49 @@ const LogWorkout = (props) => {
 
   const [exerciseState, setExerciseState] = useState(initialExerciseState);
 
+  const currentUser = firebase.auth().currentUser.uid;
+  // const currentUser = '68w6wWz8l5QJO3tDukh1fRXWYjD2';
+
+  const dbRef = firebase.firestore();
+  const userRef = dbRef.collection('users').doc(currentUser);
+  const dispatch = useDispatch();
+
+  const incrementSelectedCycleIdx = () => {
+    dispatch({ type: INCREMENT_SELECTED_CYCLE_INDEX, cycleLength });
+    userRef.update({ selectedCycleIndex: (cycleIdx + 1) % cycleLength });
```

Comment:
Can you flip these two lines. I know it works this way but I'm a bit worried that since you're using the actual selectedCycleId this might increment it 2 times in the database (even though it doesn't). It's just weird to me that it's not being re-rendered after your action completes.